lib/scrapPaperReader.js

```diff
@@ -10,4 +85,410 @@ const readPdf = async (pdfBuffer) => {
+ * relative to patch size.
+ * @param {string} filepath vertical slice of a jpeg full page
+ * @returns {string} a barcode string or null
+ */
+const _decodeJpegSegment = async (filepath, patchSize) => {
+  return new Promise((resolve, reject) => {
+    try {
+      // as barcode changes size relative to page, we should adjust patchSize. May want to guarantee barcode size and page size.
+      quagga.decodeSingle(
+        {
+          src: filepath,
+          halfSample: true,
+          locate: true,
+          patchSize: patchSize,
+          numOfWorkers: 0,
+          inputStream: {
+            size: 1920, // restrict input-size resolution to be 1920 in width (long-side)
+          },
+          decoder: {
+            readers: ['code_128_reader'],
+          },
+        },
+        (result) => {
+          if (result && result.codeResult && result.codeResult.code) {
+            resolve(result.codeResult.code);
+          } else {
+            resolve(null);
+          }
+        }
+      );
+    } catch (err) {
+      reject(err);
+    }
+  });
+};
+
+/**
+ * Splits jpeg image into vertical slices to work with Quagga barcode reader.
+ * @param {number} pageNum the page number folder that the segmented files should be placed in
+ * @param {string} jpegFilepath filepath referencing jpeg full page image
+ * @param {string} workingDir working directory where image processing occurs
+ * @returns {string} contains filepaths referencing jpeg segments
+ */
+const _segmentJpeg = async (pageNum, jpegFilepath, workingDir, logger, optionNum) => {
+  const segmentsDir = path.join(workingDir, String(pageNum));
+  const segmentsFilepath = path.join(segmentsDir, `segment-${optionNum}.jpeg`);
+  const args = [jpegFilepath].concat(segmentOptions[optionNum], [segmentsFilepath]);
+  logger.info(`Segmenting ${jpegFilepath} using imagemagick command: 
+    'convert ${args.join(' ')} `);
+  await fs.mkdir(segmentsDir, { recursive: true });
+
+  const filenames = await new Promise((resolve, reject) => {
+    imagemagick.convert(args, (err) => {
+      if (err) {
+        reject(err);
+      }
+      resolve(fs.readdir(segmentsDir));
+    });
+  });
+  const filepaths = filenames.map((filename) => {
+    return path.join(segmentsDir, filename);
+  });
+  return filepaths;
+};
+
+const _slicePageAs = async (pageNum, pdfFilePath, workingDir, extension) => {
+  return new Promise((resolve, reject) => {
+    const filename = `${pageNum}.${extension}`;
+    const filepath = path.join(workingDir, filename);
+
+    imagemagick.convert(
+      // first arg: ie. name.pdf[0] , first page starts at 0
+      [
+        `${pdfFilePath}[${pageNum}]`,
+        '-flatten',
+        '-quality',
+        '100',
+        '-adaptive-sharpen',
+        '0x3',
+        filepath,
+      ],
+      (err) => {
+        // stdout 2nd arg, but always empty
+        if (err) {
+          reject(err);
```

Comment:
You should `return` after this or else the promise will always resolve even after it rejects.

Reply:
Done.